src/gui/gui_ledger_window.hpp

```diff
@@ -1918,28 +1920,115 @@ class ledger_commodity_plupp : public tinted_image_element_base {
+			auto newest_index = economy::most_recent_price_record_index(state);
+			if(price_toggle_status[commodity.index()]) {
+				for(uint32_t i = 0; i < graph_length; ++i) {
+					auto price = state.world.commodity_get_price_record(commodity, (newest_index + economy::price_history_lenght - graph_length + i + 1) % economy::price_history_lenght);
+					if(price > max) {
+						max = price;
+					}
+				}
+			}
+		});
+
+		state.world.for_each_commodity([&](dcon::commodity_id commodity) {
+			if(!(price_toggle_status[commodity.index()])) {
+				graph_per_price[commodity.index()]->set_visible(state, false);
+			} else {
+				graph_per_price[commodity.index()]->set_visible(state, true);
+				std::vector<float> datapoints(graph_length);
+				auto newest_index = economy::most_recent_price_record_index(state);
+
+				for(uint32_t i = 0; i < graph_length; ++i) {
+					datapoints[i] = state.world.commodity_get_price_record(commodity, (newest_index + economy::price_history_lenght - graph_length + i + 1) % economy::price_history_lenght);
+				}
+				graph_per_price[commodity.index()]->set_data_points(state, datapoints, min, max);
+			}
+		});
+	}
+};
+
+
+class price_toggle_checkbox : public checkbox_button {
+public:
+	int32_t index;
+	price_toggle_checkbox(dcon::commodity_id good_id) {
+		index = good_id.index();
+	};
+	void button_action(sys::state& state) noexcept {
+		if(index < 0) return;
+		price_toggle_status[index] = !price_toggle_status[index];
+		state.game_state_updated.store(true, std::memory_order_release);
+	}
+	bool is_active(sys::state& state) noexcept {
+		if(index < 0) return false;
+		return price_toggle_status[index];
+	}
+};
+
 class ledger_commodity_price_entry : public listbox_row_element_base<dcon::commodity_id> {
 public:
 	std::unique_ptr<element_base> make_child(sys::state& state, std::string_view name, dcon::gui_def_id id) noexcept override {
 		if(name == "ledger_default_textbox") {
 			return make_element_by_type<generic_name_text<dcon::commodity_id>>(state, id);
 		} else if(name == "ledger_legend_plupp") {
 			return make_element_by_type<ledger_commodity_plupp>(state, id);
+		} else if(name == "ledger_legend_checkbox") {
+			return make_element_by_type<price_toggle_checkbox>(state, id, content);
 		} else {
 			return nullptr;
 		}
 	}
+
+	void on_update(sys::state& state) noexcept override {
+		(reinterpret_cast<price_toggle_checkbox*>(get_child_by_name(state, "ledger_legend_checkbox")))->index = content.index();
+	}
 };
+
 class ledger_commodity_price_listbox : public listbox_element_base<ledger_commodity_price_entry, dcon::commodity_id> {
 protected:
 	std::string_view get_row_element_name() override {
 		return "ledger_legend_entry";
 	}
-
 public:
 	void on_update(sys::state& state) noexcept override {
 		row_contents.clear();
-		state.world.for_each_commodity([&](dcon::commodity_id id) { row_contents.push_back(id); });
+		state.world.for_each_commodity([&](dcon::commodity_id id) {
```

Comment:
I know that you didn't write this loop. But, as a comment on the original loop, since the commodities themselves don't change over the course of the game, the list could be generated just once when the element is created. Doesn't really matter I suppose